New package: freecad-0.16

yopito opened the pull request:
runtime is fine with x86_64/glibc.
cross-build not yet tested, let's see what's Travis CI think about it.
might also rename some packages by prefixing them with "coin3-" ?

yopito commented:
rebased: fix musl (native) build for coin3 and oce packages.

Note that Travis CI is not that helpfull: some build stops because log file is 4MB, other because build time exceeded 120 minutes.

Also has troubles with cross-building, since coin3 provides a shell script `coin3-config`, so it should be in `hostmakedepend` ... might find a workaround if relying on `pkgconfig` stuff instead ?

Any help is welcome ;)

ebfe commented on common/shlibs:
```diff
@@ -2458,3 +2458,67 @@ libmarkdown.so.2 discount-2.2.0_3
+libTKXmlTObj.so.10 oce-0.17.1_1
+libTKXmlXCAF.so.10 oce-0.17.1_1
+libTKernel.so.10 oce-0.17.1_1
+libCoin.so.60 coin3-3.1.3_1
```
these probably belong into the coin3 commit

yopito replied:
my bad, thanks for pointing this.

ebfe commented on srcpkgs/gl2ps/template:
```diff
@@ -0,0 +1,30 @@
+post_install() {
+	rm ${DESTDIR}/usr/lib/*.a
+	vlicense COPYING.GL2PS
+	vcopy gl2ps.pdf    usr/share/doc/${pkgname}
```
vdoc gl2ps.pdf

ebfe replied:
and maybe move to the -devel package?

Vaelatern replied:
Yeah, the `*.a` in particular should go in the `-devel` package

Vaelatern commented on srcpkgs/coin3/template:
```diff
@@ -0,0 +1,66 @@
+ --enable-3ds-import --enable-html"
+hostmakedepends="pkg-config doxygen perl"
+makedepends="freetype-devel glu-devel zlib-devel bzip2-devel fontconfig-devel
+ freetype-devel boost-devel simage-devel"
```
`freetype-devel` is repeated here

Vaelatern commented on srcpkgs/freecad/template:
```diff
@@ -0,0 +1,60 @@
+
+	# AppHomePath is computed with binary's realpath:
+	# do not move binaries but symlink them instead.
+	mkdir ${DESTDIR}/usr/bin
```
`vmkdir usr/bin`

Vaelatern commented on srcpkgs/freecad/template:
```diff
@@ -0,0 +1,60 @@
+	done
+
+	# desktop integration
+	mkdir -p ${DESTDIR}/usr/share/applications
```
`vmkdir usr/share/applications`

Vaelatern commented on srcpkgs/freecad/template:
```diff
@@ -0,0 +1,60 @@
+		ln -s ${_inst_prefix}/data/freecad-icon-${s}.png ${DESTDIR}/${_dest_icon}/freecad.png
+		ln -s ${_inst_prefix}/data/freecad.xpm ${DESTDIR}/${_dest_icon}/freecad.xpm
+	done
+	mkdir -p ${DESTDIR}/usr/share/icons/scalable/apps
```
`vmkdir usr/share/icons/scalable/apps`

Vaelatern commented on srcpkgs/freecad/template:
```diff
@@ -0,0 +1,60 @@
+	sed -i -e "s,@_inst_prefix@,${_inst_prefix}," ${DESTDIR}/usr/share/applications/freecad.desktop
+	mkdir -p ${DESTDIR}/usr/share/appdata
+	vcopy ${FILESDIR}/freecad.appdata.xml usr/share/appdata
+	mkdir -p ${DESTDIR}/usr/share/mime/packages
```
`vmkdir`

Vaelatern commented on srcpkgs/freecad/template:
```diff
@@ -0,0 +1,60 @@
+	mkdir -p ${DESTDIR}/usr/share/applications
+	vcopy ${FILESDIR}/freecad.desktop usr/share/applications
+	sed -i -e "s,@_inst_prefix@,${_inst_prefix}," ${DESTDIR}/usr/share/applications/freecad.desktop
+	mkdir -p ${DESTDIR}/usr/share/appdata
```
`vmkdir`

Vaelatern commented on srcpkgs/libspnav/template:
```diff
@@ -0,0 +1,25 @@
+checksum=7ae4d7bb7f6a5dda28b487891e01accc856311440f582299760dace6ee5f1f93
+
+post_install() {
+	rm ${DESTDIR}/usr/lib/libspnav.a
```
Instead, in the `-devel` package, `vmove usr/lib/*.a`

yopito commented on srcpkgs/coin3/template:
```diff
@@ -0,0 +1,66 @@
+	export CPPFLAGS+="$(pkg-config --cflags freetype2)"
+	: ${configure_script:=./configure}
+	${configure_script} ${configure_args}
+}
```
by irc: test `export CPPFLAGS+="$(pkg-config --cflags freetype2)"` in the core of the template and remove `do_configure()` then

yopito commented:
rebased to fix all comments.
FIY, x86_64/musl native builds succeeded, have to test its runtimes though.

yopito commented:
Rebased: 
* small fixes
* flag packages that needs `nocross=yes` 
* added `pycompile_dirs`  directive into FreeCAD's template.

As usual, Travis CI will probably timeout because of build lenght (2+ hours) or log file (4+ MB).

I've tested the runtime with x86_64/glibc and x86_64/musl, including opening the provided samples and using the internal tests of the application (`FreeCADCmd -t 1`).

yopito commented:
rebased again, sorry: fix cross-build of libspnav

yopito commented:
rebased on top of master, no change.

karl-nilsson commented:
* FreeCAD's license is LGPL2+
https://github.com/FreeCAD/FreeCAD/blob/releases/FreeCAD-0-16/copying.lib
* SoQt is not required, and its replacement (Quarter) is already part of the freecad source tree
http://forum.freecadweb.org/viewtopic.php?f=8&t=733&start=80#p117634

Additional Requirements (starting here: https://github.com/FreeCAD/FreeCAD/blob/releases/FreeCAD-0-16/CMakeLists.txt#L568

* zlib
* boost
* PyCXX

yopito commented:
Commits just updated (squashed). See details below of what has changed.

@karl-nilsson : many thanks for your comments. Here my replies:

* FreeCAD license: fixed

* SoQt: was a dep of python-pivy package. Checking its source code shows that it uses Quarter if no other toolkit is present (SoQt, SoGTK, etc). So you're right ! Sorry for that (its build output did not notice that). So I removed the SoQt package and the corresponding dep in my python-pivy package

* FreeCAD / zlib : indirect dependency (not explicitely required), but it's present. FreeCAD cmake's output:  
    ```
...
-- Found Xerces-C: /usr/lib64/libxerces-c.so
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
    ```

* FreeCAD / boost: already present in `freecad/template`. Confirmed by cmake output:  
    ```
...
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   filesystem
--   program_options
--   regex
--   signals
--   system
--   thread
--   chrono
--   date_time
--   atomic
    ```

* FreeCAD / PyCXX: PyCXX is not (yet ?) packaged in XBPS repo, so FreeCAD uses its internal copy. Is its a problem ? cmake's output:
    ```
...
-- PyCXX found:
--   Headers:  /builddir/FreeCAD-0.16/src
--   Sources:  /builddir/FreeCAD-0.16/src/CXX
    ```

* FreeCAD and OCE/OCCT: please notice that I provide a packaging against OCE "only" ;)

yopito commented:
updated (and squash): oce build is broken on arm. will try to fix it later

yopito commented:
rebased, updated (and squashed): few updates on packaging comments (discussed with @Vaelatern on IRC).

@karl-nilsson reported (IRC) successfull package builds and basic runtime usage of FreeCAD is fine

Vaelatern commented:
Closed.
https://git.io/vKtuS
